maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/report/ConsoleOutputFileReporter.java

```diff
@@ -69,8 +69,8 @@ public void close()
         {
             try
             {
-                fileOutputStream.flush();
-                fileOutputStream.close();
+                // fileOutputStream.flush(); Will not call close on exception!
+                fileOutputStream.close(); // Will implicitly flush.
```

Comment:
Then the correct way to do that is wrap the flush in another layer of try catch (and leave a todo to use addSuppressed once baseline JVM for the code is Java 7)